### PR TITLE
Support Dynamic Shape Based on Lora, Dreambooth, Flash Attention

### PR DESCRIPTION
**Pull request by wcrzlh**
# What does this PR do?

Fixes # (issue)

Adds # (feature)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [x] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

File: examples/stable_diffusion_xl/gm/modules/diffusionmodules/model.py [hunk 1]
```diff
@@ -51,8 +51,8 @@ def construct(self, x):
         if self.with_conv:
             # x = mnp.pad(x, ((0, 0), (0, 0), (0, 1), (0, 1)))
             b, c, h, w = x.shape[0], x.shape[1], x.shape[2], x.shape[-1]
-            concat_h = ops.ones((b,c,h,1))
-            concat_w = ops.ones((b,c,1,w+1))
+            concat_h = ops.ones((b,c,h,1)).astype(x.dtype)
```
Comment (CaitinZhao) [inline examples/stable_diffusion_xl/gm/modules/diffusionmodules/model.py:54]: ops.ones((b,c,h,1), x.dtype)

File: examples/stable_diffusion_xl/configs/training/sd_xl_base_finetune_multi_aspect.yaml [hunk 2]
```diff
@@ -33,7 +33,7 @@ model:
                 use_linear_in_transformer: True
                 transformer_depth: [1, 2, 10]  # note: the first is unused (due to attn_res starting at 2) 32, 16, 8 --> 64, 32, 16
                 context_dim: 2048
-                spatial_transformer_attn_type: vanilla  # vanilla, flash-attention
+                spatial_transformer_attn_type: flash-attention  # vanilla, flash-attention
```
Comment (CaitinZhao) [inline examples/stable_diffusion_xl/configs/training/sd_xl_base_finetune_multi_aspect.yaml:36]: 不建议改

File: examples/stable_diffusion_xl/gm/modules/diffusionmodules/model.py (hunk 1 above)
Comment (wcrzlh) [reply examples/stable_diffusion_xl/gm/modules/diffusionmodules/model.py]: fixed

File: examples/stable_diffusion_xl/configs/training/sd_xl_base_finetune_multi_aspect.yaml (hunk 2 above)
Comment (wcrzlh) [reply examples/stable_diffusion_xl/configs/training/sd_xl_base_finetune_multi_aspect.yaml]: fixed

**Review by CaitinZhao — APPROVED** [review]

**Pull request merged**